Framework/MDAlgorithms/src/LoadGaussCube.cpp

```diff
@@ -0,0 +1,145 @@
+#include "MantidDataObjects/MDHistoWorkspace.h"
+#include "MantidKernel/Strings.h"
+#include "MantidMDAlgorithms/CreateMDHistoWorkspace.h"
+
+#include <fstream>
+
+using namespace Mantid::API;
+using namespace Mantid::Kernel::Strings;
+
+namespace Mantid {
+namespace MDAlgorithms {
+using Mantid::API::WorkspaceProperty;
+using Mantid::Kernel::Direction;
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(LoadGaussCube)
+
+//----------------------------------------------------------------------------------------------
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string LoadGaussCube::name() const { return "LoadGaussCube"; }
+
+/// Algorithm's version for identification. @see Algorithm::version
+int LoadGaussCube::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string LoadGaussCube::category() const { return "MDAlgorithms"; }
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string LoadGaussCube::summary() const {
+  return "Algorithm to load gauss cube files and output an 3D MDHistoWorkspace.";
+}
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void LoadGaussCube::init() {
+  auto algcreateMD = AlgorithmManager::Instance().createUnmanaged("CreateMDHistoWorkspace");
+  algcreateMD->initialize();
+
+  const std::vector<std::string> exts{".cube"};
+  declareProperty(std::make_unique<FileProperty>("Filename", "", FileProperty::Load, exts),
+                  "Path to gauss cube file (with extension .cube). Note algorithm assumes XYZ ordering.");
+
+  declareProperty(std::make_unique<WorkspaceProperty<IMDHistoWorkspace>>("OutputWorkspace", "", Direction::Output),
+                  "3D MDHistoWorkspace containing the data in the .cube file.");
+
+  copyProperty(algcreateMD, "Names");
+  copyProperty(algcreateMD, "Frames");
+  copyProperty(algcreateMD, "Units");
+}
+
+//----------------------------------------------------------------------------------------------
+/** Validate input
+ */
+std::map<std::string, std::string> LoadGaussCube::validateInputs() {
+  // check three dimensions specified
+  std::map<std::string, std::string> errors;
+  std::vector<std::string> prop_names = {"Names", "Frames", "Units"};
+  for (auto name : prop_names) {
+    std::vector<std::string> prop = getProperty(name);
+    if (prop.size() != 3) {
+      errors.emplace(name, "Property must contain three elements (workspace must have three dimensions).");
+    }
+  }
+  return errors;
+}
+
+//----------------------------------------------------------------------------------------------
+/** Execute the algorithm.
+ */
+void LoadGaussCube::exec() {
+  std::string Filename = getProperty("Filename");
+
+  // open file
+  std::ifstream in(Filename.c_str());
+  // skip header
+  for (int iline = 0; iline < 2; ++iline) {
```

Comment:
what is `iline`?

Reply:
index of line

Reply:
I would expect it to be either `indexOfLine` or `line_i` (I personally like the second one better, but thats me with my science hat on)